codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java

```diff
@@ -90,7 +90,7 @@ public CharSequence escape(CharSequence name, CharSequence value) {
         CombinedHttpHeadersImpl(HashingStrategy<CharSequence> nameHashingStrategy,
                                 ValueConverter<CharSequence> valueConverter,
                                 NameValidator<CharSequence> nameValidator,
-                                ValueValidator<CharSequence> valueValidator) {
+                                ValueValidator<CharSequence, CharSequence> valueValidator) {
```

Comment:
API Breakage

Reply:
I know there is API breakage, but I was wondering if we can still afford it given that this interface is pretty recent.

Reply:
No, we can't break applications.

Reply:
Hmm, then I need to figure out another way to add this debug information. BTW the original change of adding the validation itself is a breaking change.

Reply:
No, it isn't. The public API was preserved.

Reply:
The public API is preserved but the actual runtime behavior was changed. Prior to the validation change the header values were leniently accepted but now there is a stricter validation, which IMO is a good thing but it broke one of our applications which was adding an empty space in a header.

Now an error is thrown but it isn't clear which header is the one which has a whitespace.

Reply:
I agree the runtime behaviour changed but I still think it was an "acceptable" change due the security impact of not doing so

Reply:
I didn't look to closely but maybe what could be done is to "catch and rethrow" when this happens. The caller would then add the extra details

Reply:
@normanmaurer That was what I initially thought of doing, but it seemed hacky. I have another approach of creating another interface which is also API compatible. I will update the PR, if you like it that's fine. Otherwise will fallback to catch, change message and rethrow.

Reply:
@normanmaurer This is what I am thinking of doing,

1. Create a new interface `HeaderValueValidator<K, V>`, the benefit of this interface would be if we want to do a contextual validations of headers on the basis of what the header represents it would be possible via this interface. I feel validation of header values without having context of their name works now but seems inflexible to me.
3. Mark `ValueValidator` as deprecated but still keep it around because we can't break public APIs. During actual validation we check which of `HeaderValueValidator` or  `ValueValidator` has been provided and based on that call the appropriate validator. If a user is extending `DefaultHeaders` their implementations would still pass  `ValueValidator` and continue to function as is.

I could have made `HeaderValueValidator` extend `ValueValidator` but that pollutes the new interface with an unnecessary `validate(V value)`.